xlators/cluster/ec/src/ec-inode-write.c

```diff
@@ -755,6 +755,20 @@ ec_setxattr(call_frame_t *frame, xlator_t *this, uintptr_t target,
         }
     }
 
+    ec_t *ec = fop->xl->private;
+    int64_t val = 0;
+    int ret = dict_get_int64(fop->dict, SQUOTA_LIMIT_KEY, &val);
+    if (IS_SUCCESS(ret)) {
+        /* divide the total usage to priv->fragments */
+        int64_t new_value = val / ec->fragments;
+        ret = dict_set_int64(fop->dict, SQUOTA_LIMIT_KEY, new_value);
+        if (IS_ERROR(ret)) {
+            /* Add a trace log */
```

Comment:
IMO if this fails we need to fail the whole operation. Otherwise the brick will have an incorrect quota limit that will make quota work incorrectly.

Reply:
Done